components/src/preact/lineageFilter/fetchLineageAutocompleteList.ts

```diff
@@ -1,20 +1,116 @@
+    const [countsByLineage, lineageTree] = await Promise.all([
+        getCountsByLineage({
+            lapisUrl,
+            lapisField,
+            lapisFilter,
+            signal,
+        }),
+        getLineageTree({ lapisUrl, lapisField, signal }),
+    ]);
+
+    return Array.from(lineageTree.keys())
+        .map((lineage) => {
+            return [
+                {
+                    lineage,
+                    count: countsByLineage.get(lineage) ?? 0,
+                },
+                {
+                    lineage: `${lineage}*`,
+                    count: getCountsIncludingSublineages(lineage, lineageTree, countsByLineage),
+                },
+            ];
+        })
+        .sort((a, b) => a[0].lineage.localeCompare(b[0].lineage))
+        .flat();
+}
+
+async function getCountsByLineage({
+    lapisUrl,
+    lapisField,
     lapisFilter,
+    signal,
 }: {
-    lapis: string;
-    field: string;
+    lapisUrl: string;
+    lapisField: string;
     lapisFilter?: LapisFilter;
     signal?: AbortSignal;
 }) {
-    const fetchAggregatedOperator = new FetchAggregatedOperator<Record<string, string>>(lapisFilter ?? {}, [field]);
+    const fetchAggregatedOperator = new FetchAggregatedOperator<Record<string, string>>(lapisFilter ?? {}, [
+        lapisField,
+    ]);
+
+    const countsByLineageArray = (await fetchAggregatedOperator.evaluate(lapisUrl, signal)).content;
+    return countsByLineageArray.reduce(
+        (acc, value) => acc.set(value[lapisField], value.count),
+        new Map<string, number>(),
+    );
+}
+
+async function getLineageTree({
+    lapisUrl,
+    lapisField,
+    signal,
+}: {
+    lapisUrl: string;
+    lapisField: string;
+    signal?: AbortSignal;
+}) {
+    const lineageDefinitions = await fetchLineageDefinition({ lapisUrl, lapisField, signal });
+
+    const lineageTree = new Map<string, { children: string[] }>();
+
+    Object.entries(lineageDefinitions).forEach(([lineage, definition]) => {
+        lineageTree.set(lineage, { children: [] });
+
+        definition.parents?.forEach((parent) => {
+            const parentChildren = lineageTree.get(parent)?.children;
+
+            const newParentChildren = parentChildren ? [...parentChildren, lineage] : [lineage];
+
+            lineageTree.set(parent, { children: newParentChildren });
+        });
+    });
+
+    return lineageTree;
+}
+
+function getCountsIncludingSublineages(
+    lineage: string,
+    lineageTree: Map<string, { children: string[] }>,
+    countsByLineage: Map<string, number>,
+): number {
+    const descendents = new Set(getAllDescendents(lineage, lineageTree));
+
+    const countOfChildren = [...descendents].reduce((sum, child) => {
+        return sum + (countsByLineage.get(child) ?? 0);
+    }, 0);
+    const countLineage = countsByLineage.get(lineage) ?? 0;
+
+    return countOfChildren + countLineage;
+}
+
+function getAllDescendents(lineage: string, lineageTree: Map<string, { children: string[] }>): string[] {
```

Comment:
descendAnt seems the more common spelling (rather than descendEnt) according to a quick search.
```suggestion
function getAllDescendants(lineage: string, lineageTree: Map<string, { children: string[] }>): string[] {
```